Sources/Plasma/PubUtilLib/plNetCommon/plNetObjectDebugger.cpp

```diff
@@ -52,48 +52,25 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #include "plResMgr/plKeyFinder.h"
 #include "plStatusLog/plStatusLog.h"
 
-plNetObjectDebugger::DebugObject::DebugObject(const char* objName, plLocation& loc, uint32_t flags) :
-fLoc(loc),
-fFlags(flags)
-{
-    std::string tmp = objName;
-    hsStrLower((char*)tmp.c_str());
-    fObjName = tmp;
-}
-
 //
 // return true if string matches objName according to flags
 //
-bool plNetObjectDebugger::DebugObject::StringMatches(const char* str) const
+bool plNetObjectDebugger::DebugObject::StringMatches(const ST::string& str) const
 {
-    if (!str)
+    if (str.empty())
         return false;
 
     if (fFlags & kExactStringMatch)
-        return !stricmp(str, fObjName.c_str());
+        return str.compare(fObjName, ST::case_insensitive) == 0;
 
     if (fFlags & kEndStringMatch)
-    {
-        int len=strlen(str);
-        if (len>fObjName.size())
-            return false;
-        return !stricmp(str, fObjName.c_str()+fObjName.size()-len);
-    }
+        return str.ends_with(fObjName, ST::case_insensitive);
```

Comment:
The observant reviewer may notice that the old code actually performed the equivalent of `fObjName.ends_with(str, ...)`.  However, since the pattern specified in `fObjName` was created as "*foo", this didn't make much sense (and the SubStringMatch case below also fits that `fObjName` should be the search criteria, not the text to search in).  Therefore, I've corrected it to match what I believe was the intended behavior.